src/App.js

```diff
@@ -69,14 +71,17 @@ function App() {
                     PickerStateProvider,
                     EnvironmentProvider,
                     CustomStatusBarAndBackgroundContextProvider,
+                    ActionSheetAwareScrollView.ActionSheetAwareScrollViewProvider,
                 ]}
             >
-                <CustomStatusBarAndBackground />
-                <ErrorBoundary errorMessage="NewExpensify crash caught by error boundary">
-                    <ColorSchemeWrapper>
-                        <Expensify />
-                    </ColorSchemeWrapper>
-                </ErrorBoundary>
+                <KeyboardProvider enabled={Platform.OS === "ios"}>
```

Comment:
I notice you're enabling this only on iOS, but this issue occurs the same on Android as it does on iOS. So I think we'd like to see a consistent approach to keyboard avoidance on all platforms.

Currently, I think we use an unintuitive mix of native and JS-layer keyboard avoidance, and it's inconsistent between platforms.

Reply:
Also, should this be inside the error boundary?

Reply:
> Also, should this be inside the error boundary?

I added it to the level of all providers - the reason why it's a separate JSX element because I specify `props` (enabled) and right now `ComposeProviders` doesn't have an ability to specify additional props. And as I can see all providers are not under `ErrorBoundary`.

Would you mind to give a brief explanation why do you think it's worth to put this element under `ErrorBoundary`?

> Currently, I think we use an unintuitive mix of native and JS-layer keyboard avoidance, and it's inconsistent between platforms.

Yes, because an initial PR supported only iOS (from PR description: _This PR is iOS only for now. Other platforms are not affected. Also, in this explanation, we are talking about iOS only since on Android, iOS web and Android Web keyboard handling works differently_.)

But I agree with you that it would be good to have cross-platform code that handles keyboard interactions. And we actually wanted to suggest to do that, but our vision was to merge this PR first (because it's quite old and contains a lot of changes) and then add Android support in next PRs. What do you think about it?

Reply:
@kirillzyusko lets create a new `KeyboardHandlerProvider` in components, that returns the <KeyboardProvider with the `enabled` prop set.
Then we can pass it to `ComposeProviders` so it's in line with all the other providers